### PR TITLE
[rllib] Fix error messages and example for dataset writer

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Currently the error message and example refer to a field `type` that is actually `format`.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by krfricke** [comment]
Btw @gjoliver, it seems like final batches are not written because `max_num_samples_per_file` is never reached.

Thus in the example `"max_num_samples_per_file": 5000` has to be passed as well.

Is this something we can fix? I.e. is there a deconstructor/cleanup method we can use for this?

File: rllib/offline/dataset_writer.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def __init__(
         output_config: Dict = ioctx.output_config
         assert (
             "format" in output_config
-        ), "output_config.type must be specified when using Dataset output."
+        ), "output_config.format must be specified when using Dataset output."
```
Comment (gjoliver) [inline rllib/offline/dataset_writer.py:47]: ah thanks for catching this. went through iterations ...

File: rllib/tuned_examples/marwil/cartpole-bc.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # To generate training data, first run:
 # $ ./train.py --run=PPO --env=CartPole-v0 \
 #      --stop='{"timesteps_total": 50000}' \
-#      --config='{"output": "dataset", "output_config": {"type": "json", "path": "/tmp/out"}, "batch_mode": "complete_episodes"}'
+#      --config='{"output": "dataset", "output_config": {"format": "json", "path": "/tmp/out", "max_num_samples_per_file": 1}, "batch_mode": "complete_episodes"}'
```
Comment (gjoliver) [inline rllib/tuned_examples/marwil/cartpole-bc.yaml:4]: just to double check, you want to write 1 sample per file??

Comment (krfricke) [reply rllib/tuned_examples/marwil/cartpole-bc.yaml:4]: Well 1 sample = 1 episode here so usually has a few hundred timesteps. When I ran this I ended up with 39 files.

**Review by gjoliver — APPROVED** [review]

File: rllib/tuned_examples/marwil/cartpole-bc.yaml (hunk 2 above)
Comment (krfricke) [reply rllib/tuned_examples/marwil/cartpole-bc.yaml:4]: The main reason is that currently remaining samples are not written at all - they are only written when at least `max_num_samples_per_file` samples are reached during training. I.e. we don't flush remaining results when training is over and I didn't see a great way to make that happen

Comment (gjoliver) [reply rllib/tuned_examples/marwil/cartpole-bc.yaml:4]: oh this is pretty bad. please add a Note or TODO in dataset_writer.py
this is another case where we are using a workaround to fix underlying problems. and per eric's message, we shouldn't do this long-term.
let's intercept a signal somewhere and flush the last shard before things go down.

Comment (krfricke) [reply rllib/tuned_examples/marwil/cartpole-bc.yaml:4]: Ah yeah that's what I mentioned in my first comment to this PR. Ok, will add a todo in the file. I've tried to flush on `__del__` but that doesn't seem to work with actors, so no quick fix. Probably we need to call some kind of deconstructor/cleanup method manually

Comment (gjoliver) [reply rllib/tuned_examples/marwil/cartpole-bc.yaml:4]: ah, it all comes back. I didn't understand what you meant at the time :)
cool, interesting topic ...

**Review by gjoliver — APPROVED** [review]
feel free to merge btw.

**Pull request merged**